docs/developer-guide/index.md

```diff
@@ -256,19 +256,44 @@ the steps below.
 1. Install [kustomize](https://github.com/kubernetes-sigs/kustomize)
 1. Install [tilt](https://docs.tilt.dev/install.html)
 1. Create a local Kubernetes cluster
-
-To start up your Tilt development environment, run
-
-```shell
-tilt up
-```
-
-at the root of your local NFD codebase. Tilt will start a web interface in the
-localhost and port 10350. From the web interface, you are able to see how NFD worker
-and master are progressing,  watch their build and runtime logs. Once your code changes
-are saved locally, Tilt will notice it and re-build the container image from the
-current code, push the image to the registry and re-deploy NFD pods with the latest
-container image.
+    - Create registry first through the command
+      ```
+      cat <<EOF | ctlptl apply -f -
+      apiVersion: ctlptl.dev/v1alpha1
+      kind: Registry
+      name: ctlptl-registry
+      port: 5005
+      ```
+      or
+      ```
+      ctlptl create registry ctlptl-registry --port=5005
+      ```
+    - Create a Kubernetes cluster, which will be deployed onto controller node and woker nodes running inside docker containers. Make sure mounting necessary host folders to worker nodes for feature discovery. For example, if k3d is used for cluster creation (please note that the mounted folders for feature discovering should be set to read only in production environment): 
+      ```
+      cat <<EOF | ctlptl apply -f -
+      apiVersion: ctlptl.dev/v1alpha1
+      kind: Cluster
+      registry: ctlptl-registry
+      product: k3d
+      k3d:
+        v1alpha5Simple:
+          volumes:
+            - volume: /boot:/boot
+              nodeFilters:
+                - server:0
+                - agent:*
+            - volume: /lib:/lib
+              nodeFilters:
+                - server:0
+                - agent:*
+      EOF
+      ```
```

Comment:
Thanks for the example! I believe we should avoid including a specific cluster installation method since it ties us to particular installers, which aren't standardized. For instance, `v1alpha5Simple` embeds k3d-specific configuration. Could we instead add a note highlighting the directories that need to be mounted?

Reply:
sounds good